src/pkg/immutabletag/manager.go

```diff
@@ -0,0 +1,110 @@
+package immutabletag
+
+import (
+	"encoding/json"
+	"github.com/goharbor/harbor/src/pkg/immutabletag/dao"
+	dao_model "github.com/goharbor/harbor/src/pkg/immutabletag/dao/model"
+	"github.com/goharbor/harbor/src/pkg/immutabletag/model"
+)
+
+var (
+	// Mgr is a global variable for the default immutablerule manager implementation
+	Mgr = NewDefaultRuleManager()
+)
+
+// Manager ...
+type Manager interface {
+	// CreateImmutableRule creates the Immutable Rule
+	CreateImmutableRule(m *model.Metadata) (int64, error)
+	// UpdateImmutableRule update the immutable rules
+	UpdateImmutableRule(projectID int64, ir *model.Metadata) (int64, error)
+	// EnableImmutableRule enable/disable immutable rules
+	EnableImmutableRule(id int64, enabled bool) (int64, error)
+	// GetImmutableRule get immutable rule
+	GetImmutableRule(id int64) (*model.Metadata, error)
+	// QueryImmutableRuleByProjectID get all immutable rule by project
+	QueryImmutableRuleByProjectID(projectID int64) ([]model.Metadata, error)
+	// QueryEnabledImmutableRuleByProjectID get all enabled immutable rule by project
+	QueryEnabledImmutableRuleByProjectID(projectID int64) ([]model.Metadata, error)
+	// DeleteImmutableRule delete the immutable rule
+	DeleteImmutableRule(id int64) (int64, error)
+}
+
+type defaultRuleManager struct {
+	dao dao.ImmutableRuleDao
+}
+
+func (drm *defaultRuleManager) CreateImmutableRule(ir *model.Metadata) (int64, error) {
+	daoRule := &dao_model.ImmutableRule{}
+	daoRule.Enabled = !ir.Disabled
+	daoRule.ProjectID = ir.ProjectID
+	data, _ := json.Marshal(ir)
+	daoRule.TagFilter = string(data)
+	return drm.dao.CreateImmutableRule(daoRule)
+}
+
+func (drm *defaultRuleManager) UpdateImmutableRule(projectID int64, ir *model.Metadata) (int64, error) {
+	daoRule := &dao_model.ImmutableRule{}
+	data, _ := json.Marshal(ir)
+	daoRule.TagFilter = string(data)
+	return drm.dao.UpdateImmutableRule(projectID, daoRule)
+}
+
+func (drm *defaultRuleManager) EnableImmutableRule(id int64, enabled bool) (int64, error) {
```

Comment:
the name is confusing? If I want to disable any of rule, I still have to call EnableImmutableRule()